src/python/zquantum/core/history/recorder.py

```diff
@@ -143,31 +145,27 @@ def recorder(
     save_condition: SaveCondition = always,
 ) -> ArtifactRecorderWithGradient:
     """The recorder function: variant for callables with gradient and storing artifacts."""
-    pass
```

Comment:
I'm not 100% whether it will work without `pass` .
Also I think that having them here is more explicit and increases readability.

Reply:
Python parser needs `pass` only in situations where a block has no other body. Here we have a docstring, so the parser sees the indentation, so we don't need the `pass`

Reply:
Good to know :)